networkx/algorithms/shortest_paths/weighted.py

```diff
@@ -857,36 +857,36 @@ def _dijkstra_multisource(
         heappush(fringe, (0, next(c), source))
     while fringe:
         (d, _, v) = heappop(fringe)
-        if v in dist:
+        v_dist = dist.setdefault(v, d)
+        if v_dist < d:
```

Comment:
Doesn't `v_dist` have to be `< d` if `v in dist`? 
Why not use:
```python
if v in dist:
    continue
dist[v] = v_dist = d
```
This also avoids `setdefault` which I've been advised (by cpython devs) to avoid, though function calls are speeding up relative to operations.